esp-hal/src/uart.rs

```diff
@@ -285,6 +285,9 @@ pub enum Error {
     /// match the expected parity configuration.
     /// with the `async` feature.
     RxParityError,
+
+    /// This operation requires blocking behavior to complete
+    WouldBlock,
```

Comment:
I don't think we need this error. Usual `Write` semantics just return a number of bytes written, and this PR already refactors reads into Options in most places. You're pretty much only using this error where a `usize` would be enough, or where you still use the `nb` crate.